compactor2/src/components/target_level/hot_cold.rs

```diff
@@ -0,0 +1,105 @@
+use std::fmt::Display;
+
+use data_types::CompactionLevel;
+
+use crate::components::level_filter::{one_level::OneLevelFilter, LevelFilter};
+
+use super::TargetLevelDetection;
+
+#[derive(Debug)]
+pub struct HotColdTargetLevelDetection {
+    level_filter: OneLevelFilter,
+}
```

Comment:
```suggestion
#[derive(Debug)]
pub struct HotColdTargetLevelDetection<T> where T: LevelFilter {
    level_filter: T,
}
```

Because otherwise you could just inline the level filter into this file and not create a component. A component only makes sense if it is also used in a abstract way (and potentially can have multiple implementations).